### PR TITLE
Incremental backup support

### PR DESCRIPTION
**Pull request by shdeshpa07**
## Jira 

* [OADP-4820](https://issues.redhat.com/browse/OADP-4820)

Added a note to express support for incremental backups 

##  Version

* OCP 4.13 → OCP 4.17

## Preview

*  [Data Mover](https://81606--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/about-oadp-1-3-data-mover.html#oadp-about-incremental-backup-support_about-oadp-1-3-data-mover)

* [OpenShift Virtualization](https://81606--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.html#oadp-about-incremental-backup-support_installing-oadp-kubevirt)

## QE Review

* [x] QE has approved this change.

### CODE REVIEW

**Comment by shdeshpa07** [comment]
@anarnold97  -  PR is ready for your review please. Thanks.

File: snippets/oadp-incremental-backup-support-snippet.adoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+:_mod-docs-content-type: SNIPPET
+
+[NOTE]
+====
+{oadp-short} supports incremental backups of `block` and `Filesystem` persistent volumes for both containerized, and {VirtProductName} workloads. 
+
+For containerized workloads, you can perform incremental backups by using the {oadp-short} file system backup, Container Storage Interface (CSI), or CSI Data Mover.
+
+For {VirtProductName} workloads, you can perform incremental backups by using either CSI, or CSI Data Mover. CSI Data Mover backups use Kopia's splitter and hashing features  to incrementally backup a virtual machine to an off-cluster storage location.
```
Comment (anarnold97) [inline snippets/oadp-incremental-backup-support-snippet.adoc:13]:
```suggestion
For {VirtProductName} workloads, you can perform incremental backups by using either CSI, or CSI Data Mover. CSI Data Mover backups use Kopia's splitter and hashing features  to incrementally back up a virtual machine to an off-cluster storage location.
```

**Comment by weshayutin** [comment]
Sorry, @sseago reminded me we should add a note that volumeMode: Block volumes must use CSI DataMover backups, filesystem backups will not work and are not supported.

**Comment by shdeshpa07** [comment]
> Sorry, @sseago reminded me we should add a note that volumeMode: Block volumes must use CSI DataMover backups, filesystem backups will not work and are not supported.
> 
> https://redhat-internal.slack.com/archives/C0144ECKUJ0/p1726060435885529

@weshayutin  - I have reworded the note based on your and Scott's review. Could you please review again? Thanks.

**Comment by shdeshpa07** [comment]
@PrasadJoshi12  - Please review the PR. Thanks.

**Review by sseago — APPROVED** [review]

**Review by weshayutin — APPROVED** [review]
/LGTM thank you!

File: snippets/oadp-incremental-backup-support-snippet.adoc [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+:_mod-docs-content-type: SNIPPET
+
+[NOTE]
+====
+{oadp-short} supports incremental backups of `block` and `Filesystem` persistent volumes for both containerized, and {VirtProductName} workloads. 
+
+For `block` volumes, you can use the Container Storage Interface (CSI) Data Mover to perform incremental backups. File system backup is not supported for `block` volumes.
+
+For `Filesystem` volumes, you can use {oadp-short} file system backup, Container Storage Interface (CSI), or CSI Data Mover for incremental backups.
```
Comment (PrasadJoshi12) [inline snippets/oadp-incremental-backup-support-snippet.adoc:13]: I think we should remove the filesystem backup word from here as it's not supported.  

```suggestion
For `Filesystem` volumes, you can use {oadp-short} Container Storage Interface (CSI), or CSI Data Mover for incremental backups. 
```

**Comment by shdeshpa07** [comment]
> OK.. I think the thread was lost along the way.. Here's our dev version of what we want to communicate to customers. [openshift/oadp-operator#1519](https://github.com/openshift/oadp-operator/pull/1519) Hit us in office hours to discuss.

Thank you Wes for the information. I will work on the changes and get back with the updated PR for review, Thanks.

**Comment by shdeshpa07** [comment]
@weshayutin  @PrasadJoshi12  - I have incorporated the feedback. Could you please review? Thanks.

**Review by weshayutin — APPROVED** [review]
/LGTM

File: modules/oadp-incremental-backup-support.adoc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+| Filesystem | N ^[3]^ | N ^[3]^ | S ^[1]^ | S ^[1]^, I ^[2]^
+| Block | N ^[3]^ | N ^[3]^ | S ^[1]^ | S ^[1]^, I ^[2]^
+|===
+
+[.small]
+--
+1. Backup supported
+2. Incremental backup supported
+3. Not supported
+--
```
Comment (snarayan-redhat) [inline modules/oadp-incremental-backup-support.adoc:32]: Refer to the [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#alternative-footnote-styling-in-tables) for footnotes and table formatting.

**Review by snarayan-redhat — COMMENTED** [review]
Good work :) !
Left a comment that you might want to fix.

File: modules/oadp-incremental-backup-support.adoc (hunk 3 above)
Comment (shdeshpa07) [reply modules/oadp-incremental-backup-support.adoc]: The table is majorly compliant apart from the [.small] style. I have seen the [.small] style at a couple of places in OADP and it works fine so far. If I remove the [.small] style, the footnote descriptions appear in a font bigger than the table contents. Thanks.

**Pull request merged**